pkg/oc/admin/migrate/scc/scc.go

```diff
@@ -0,0 +1,158 @@
+	rbacv1helper "k8s.io/kubernetes/pkg/apis/rbac/v1"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	cmdutil "github.com/openshift/origin/pkg/cmd/util"
+	"github.com/openshift/origin/pkg/oc/cli/util/clientcmd"
+	securityapi "github.com/openshift/origin/pkg/security/apis/security"
+	securitytypedclient "github.com/openshift/origin/pkg/security/generated/internalclientset/typed/security/internalversion"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	internalMigrateSCCShort = "Converts SCCs to similar PSPs"
+	internalMigrateSCCLong  = templates.LongDesc(`
+		Converts SecurityContextConstraints to equivalent PodSecurityPolicy objects.
+
+		NOTE: it does not modify anything by default.`)
+
+	internalMigrateSCCExample = templates.Examples(`
+		# Output to the console PodSecurityPolicy objects
+		%[1]s`)
+)
+
+type migrateSCCOptions struct {
+	sccClient securitytypedclient.SecurityContextConstraintsInterface
+}
+
+func NewCmdMigrateSCC(name, fullName string, f *clientcmd.Factory, in io.Reader, out, errout io.Writer) *cobra.Command {
+	options := &migrateSCCOptions{}
+	cmd := &cobra.Command{
+		Use:     name,
+		Short:   internalMigrateSCCShort,
+		Long:    internalMigrateSCCLong,
+		Example: fmt.Sprintf(internalMigrateSCCExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(options.Complete(name, f, cmd, args))
+			kcmdutil.CheckErr(options.Validate())
+			kcmdutil.CheckErr(options.Run(cmd, f, out))
+		},
+	}
+	kcmdutil.AddPrinterFlags(cmd)
+	return cmd
+}
+
+func (o *migrateSCCOptions) Complete(name string, f *clientcmd.Factory, c *cobra.Command, args []string) error {
+	securityClient, err := f.OpenshiftInternalSecurityClient()
+	if err != nil {
+		return err
+	}
+	o.sccClient = securityClient.Security().SecurityContextConstraints()
+	return nil
+}
+
+func (o migrateSCCOptions) Validate() error {
+	if o.sccClient == nil {
+		return errors.New("a SCC client is required")
+	}
+	return nil
+}
+
+func (o migrateSCCOptions) Run(cmd *cobra.Command, f *clientcmd.Factory, out io.Writer) error {
+	sccs, err := o.sccClient.List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	list := &kapi.List{}
+	for _, scc := range sccs.Items {
+		list.Items = append(list.Items, convertSccToPsp(&scc))
+		list.Items = append(list.Items, convertSccToClusterRole(&scc))
+		binding := convertSccToClusterRoleBinding(&scc)
+		if binding != nil {
+			list.Items = append(list.Items, binding)
+		}
+	}
+
+	mapper, _ := f.Object()
+	fn := cmdutil.VersionedPrintObject(f.PrintObject, cmd, mapper, out)
+	if err := fn(list); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func convertSccToPsp(scc *securityapi.SecurityContextConstraints) *extensions.PodSecurityPolicy {
+	// TODO: convert SCC.Priority *int32 (ERR: PSP doesn't have such field)
+	// TODO: convert SCC.AllowHostPorts bool (TODO: how to convert bool to []HostPortRange{Min:, Max:}?)
+	// TODO: handle SCC.Volumes: ["none"]
```

Comment:
SCC `volumes: ["none"]` == PSP `volumes: []`

Reply:
Are you sure?

Because empty list in PSP means the opposite thing -- everything is forbidden:
https://github.com/kubernetes/kubernetes/blob/c74c826452fb03d2b876238182a3926549c4f9b0/pkg/apis/extensions/types.go#L821-L824

Reply:
that doc is definitely wrong, and is a likely copy/paste from SCC. to allow all in PSP, you need `volumes:["*"]`

https://github.com/kubernetes/kubernetes/blob/d112ffc2eb339c43f1db80c00b5b879008449c2d/pkg/security/podsecuritypolicy/provider.go#L212-L227

https://github.com/kubernetes/kubernetes/blob/b40581c14e4aeeb64625cf4a252c0a5d6abbcff5/pkg/security/podsecuritypolicy/util/util.go#L146-L164

Reply:
> that doc is definitely wrong, and is a likely copy/paste from SCC

Ok, I'll add this fix to https://github.com/kubernetes/kubernetes/pull/60792 then